src/Compilers/CSharp/Portable/Symbols/Source/SourceModuleSymbol.cs

```diff
@@ -501,29 +501,43 @@ internal override void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArgu
             if (attribute.IsTargetAttribute(this, AttributeDescription.DefaultCharSetAttribute))
             {
                 CharSet charSet = attribute.GetConstructorArgument<CharSet>(0, SpecialType.System_Enum);
-                if (!ModuleWellKnownAttributeData.IsValidCharSet(charSet))
+                if (!CommonModuleWellKnownAttributeData.IsValidCharSet(charSet))
                 {
                     CSharpSyntaxNode attributeArgumentSyntax = attribute.GetAttributeArgumentSyntax(0, arguments.AttributeSyntaxOpt);
                     arguments.Diagnostics.Add(ErrorCode.ERR_InvalidAttributeArgument, attributeArgumentSyntax.Location, arguments.AttributeSyntaxOpt.GetErrorDisplayName());
                 }
                 else
                 {
-                    arguments.GetOrCreateData<ModuleWellKnownAttributeData>().DefaultCharacterSet = charSet;
+                    arguments.GetOrCreateData<CommonModuleWellKnownAttributeData>().DefaultCharacterSet = charSet;
                 }
             }
             else if (attribute.IsTargetAttribute(this, AttributeDescription.NonNullTypesAttribute))
             {
-                bool value = attribute.GetConstructorArgument<bool>(0, SpecialType.System_Boolean);
-                arguments.GetOrCreateData<ModuleWellKnownAttributeData>().NonNullTypes = value;
+                // NonNullTypesAttribute should not be set explicitly.
+                arguments.Diagnostics.Add(ErrorCode.ERR_ExplicitNonNullTypesAttribute, arguments.AttributeSyntaxOpt.Location);
             }
         }
 
         public override bool? NonNullTypes
         {
             get
             {
-               var data = GetDecodedWellKnownAttributeData();
-               return data?.NonNullTypes;
+                string id = "CS" + ((int)ErrorCode.WRN_PragmaNonNullTypes).ToString("0000");
```

Comment:
More general question: do we really not have a better way of getting the diagnostic option given an ErrorCode? This seems like it's tying SourceModuleSymbol to how error codes are formatted, which doesn't seem like a good dependency to me.

Reply:
> More general question: do we really not have a better way of getting the diagnostic option given an ErrorCode? This seems like it's tying SourceModuleSymbol to how error codes are formatted, which doesn't seem like a good dependency to me.

I believe this is the way command line parser encodes warnings in the dictionaries. In any case this is just a temporary hack until we have dedicated command line switch and directive.

---
In reply to: [224177565](https://github.com/dotnet/roslyn/pull/30387#discussion_r224177565) [](ancestors = 224177565)